### PR TITLE
debt: add things to main entry point that people need

### PR DESCRIPTION
**Pull request by davidjgoss**
Much of the list from https://github.com/cucumber/cucumber-js/issues/1646#issue-863352190. Most of the rest already available under formatter helpers, probably just need some manual editing in those projects.

### CODE REVIEW

File: src/support_code_library_builder/types.ts [hunk 1]
```diff
@@ -25,12 +25,12 @@ export interface ITestStepHookParameter {
 
 export type TestCaseHookFunction<WorldType> = (
   this: WorldType,
-  arg?: ITestCaseHookParameter
+  arg: ITestCaseHookParameter
```
Comment (davidjgoss) [inline src/support_code_library_builder/types.ts:28]: This was a mistake from #1690. The argument isn't optional - we always pass it in - it's just up to the user-authored code whether it does anything with it.

**Comment by aurelien-reeves** [comment]
Looks good.

I wonder: is there a way to reproduce through a test the issue reported on #1646?

**Comment by davidjgoss** [comment]
>  wonder: is there a way to reproduce through a test the issue reported on #1646?

Yeah, I added a feature test in https://github.com/cucumber/cucumber-js/pull/1649 that hits the error when trying to do deep imports after the ESM change. Still working on if/how to handle that.

**Comment by aurelien-reeves** [comment]
> 
> 
> > wonder: is there a way to reproduce through a test the issue reported on #1646?
> 
> Yeah, I added a feature test in #1649 that hits the error when trying to do deep imports after the ESM change. Still working on if/how to handle that.

Great!

**Review by aurelien-reeves — APPROVED** [review]

**Pull request merged**